### ISSUE TITLE
Ipv4 and/or ipv6

### ISSUE THREAD

**Issue by moabile** [linuxserver/docker-duckdns#39]
[linuxserverurl]: https://linuxserver.io
[![linuxserver.io](https://raw.githubusercontent.com/linuxserver/docker-templates/master/linuxserver.io/img/linuxserver_medium.png)][linuxserverurl]

------------------------------

## Desired Behavior
a feature to choose ipv4 and/or ipv6 because for instance my ipv4 is not public(nat) whereas my ipv6 is public so I do not want to update my ipv4 because it causes wrong address

## Current Behavior
ipv4 cannot be disabled
ipv6 can be disabled (i think)
but this does not solve the dual stack lite issue

**Comment by aptalca** [comment]
There is no ipv6 support here, only ipv4

Is your ipv6 address dynamic?

**Comment by jsloane** [comment]
IPv6 support would be great, so IPv6 clients can use the same domain name.

**Comment by psbankar** [comment]
> There is no ipv6 support here, only ipv4
> 
> Is your ipv6 address dynamic?

Yes for me the IPv6 is dynamic. So an option for choosing just IPv6 would be great.

**Comment by Diluka** [comment]
ipv4 has been exhausted, now it is difficult to get ipv4 address, need to get the function of updating ipv6 address.

**Comment by ukkopahis** [comment]
Do you have a working example of the curl-command URL (with parameters) that will do only the IPv6 update?

**Comment by Diluka** [comment]
`curl -L https://duckdns.org/update/<your domain>/<your token>/$(curl ip.me);`

**Comment by ukkopahis** [comment]
`curl ip.me` reliably returns an ipv6 address for you? For me it returns an ipv4.

**Comment by Diluka** [comment]
~~it's random~~
or use `curl -6 ip.me`

**Comment by m00nwtchr** [comment]
You just have to manually query an IPv6 address (because DuckDNS does not autodetect it) if an env var for IPv6 IP is set as a configuration. All you need is a service which returns IPv6 addresses.
This page should be helpful: https://ip6.me/
See: https://www.duckdns.org/spec.jsp for information on how to set a ipv6 address via http query.
(use `ip` to only use IPv4 or IPv6, or `ipv6` to use both)
(IMO it should set both the IPv4 and IPv6 unless explicitly requested to only set one)

**Comment by ukkopahis** [comment]
Could someone with only an IPv6 and someone with both an IPv4 and v6 verify that you reliably get your IPv6 address when you execute:
```sh
curl -sS6 https://ip6only.me/api/ | grep -m 1 IPv6 | awk 'BEGIN{FS=","}{print $2}'
```

If this produces any errors, please report them too.

ps. if you only have IPv4 available, you should get the error `curl: (7) Couldn't connect to server`

**Comment by Tooa** [comment]
> Could someone with only an IPv6 and someone with both an IPv4 and v6 verify that you reliably get your IPv6 address when you execute

Hi 👋,

I only have a public IPv6 WAN address (DS-lite) and the command yields the correct IP address in my case.

**Comment by Tooa** [comment]
> Could someone with only an IPv6 and someone with both an IPv4 and v6 verify that you reliably get your IPv6 address when you execute

Hi 👋,

I only have a public IPv6 WAN address (DS-lite) and the command yields the correct IP address in my case.

**Comment by ukkopahis** [comment]
~~Is using ip6only.me required, or would a clearing of records and an IPv6 request to duckdns work to:~~
```sh
export SUBDOMAINS=foo
export TOKEN=xxx-yyy-zzz
curl -sS "https://www.duckdns.org/update?domains=$SUBDOMAINS&token=$TOKEN&clear=true&verbose=true"
curl -sS6 "https://www.duckdns.org/update?domains=$SUBDOMAINS&token=$TOKEN&verbose=true"
```
~~Those with an IPv6 could test and report if this results in only an IPv6 record being set into duckdns?~~ 
EDIT: not supported.

---

As a design consideration, how should the added docker image environment parameter be defined? Maybe something like:
| Parameter | Function |
| --- | --- |
| `IPV6=false` | The default (`false`) results in only IPv4 being updated. Set to `true` to update both IPv4 and IPv6. Set to `only` to clear the IPv4 and only update the IPv6. |

**Comment by m00nwtchr** [comment]
@ukkopahis  On their site it says they don't automatically detect IPv6.

https://www.duckdns.org/faqs.jsp

> Q: why can't you detect IPv6 addresses?
> 
> A: our service is hosted in AWS, they [do not support IPv6 on Elastic Load Balancers](http://docs.aws.amazon.com/ElasticLoadBalancing/latest/DeveloperGuide/elb-internet-facing-load-balancers.html) for the account type we have (VPC), when amazon do support it, we will start detecting it.

**Comment by ukkopahis** [comment]
> On their site it says they don't automatically detect IPv6

Oops, my bad. `curl https://ip6only.me/api/` for the IPv6 it is.

Request for comments about parameter name and values still stands.

**Comment by ukkopahis** [comment]
@aptalca Do you feel IPv6 support can be added into the base container or should it be implemented as a mod?

**Comment by m00nwtchr** [comment]
bump

**Comment by aptalca** [comment]
> Do you feel IPv6 support can be added into the base container or should it be implemented as a mod?

Depends on the complexity. If it's complex, mod is better

**Comment by qq380495995** [comment]
我不需要更新路由器或网关的ipv6，
我需要更新服务器本地的ipv6，
请给出配置参数，谢谢。
I don't need to update the IPv6 of my router or gateway,
I need to update the IPv6 local to the server,
Please give the configuration parameters, thank you.

**Comment by codyc1515** [comment]
Still an issue

**Comment by chiefdeputy** [comment]
I also have a dynamic IPV6 and am in need of this.

> As a design consideration, how should the added docker image environment parameter be defined? Maybe something like:
> Parameter 	Function
> `IPV6=false` 	The default (`false`) results in only IPv4 being updated. Set to `true` to update both IPv4 and IPv6. Set to `only` to clear the IPv4 and only update the IPv6.

Why not two variables?
* IPV4=true
* IPV6=false

This way it's completely backwards compatible and we have all use cases covered.

Are there any other blocking points?

**Comment by drizuid** [comment]
We have no plans to create a PR for this, it one of you facing the issue would like to do so, please submit a PR. If the PR is extremely complex, create it as a docker mod, if it's simple and doesn't negatively impact ipv4, a PR directly to this will work. If none of you are willing to do the work, this will not be added. 

tl;dr this issue is stale because the individuals who want the feature haven't provided functional code to support it.

**Comment by drizuid** [comment]
closing this as no one has shown any interest in creating a PR.
